trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/rest/TrinoS3ProxyClient.java

```diff
@@ -86,12 +87,15 @@ public void proxyRequest(SigningMetadata signingMetadata, ContainerRequest reque
             throw new WebApplicationException(Response.Status.BAD_REQUEST);
         }
 
-        // modify the source request headers for real values needed for the real AWS request
-        MultivaluedMap<String, String> realRequestHeaders = new MultivaluedHashMap<>(request.getRequestHeaders());
-        // we don't use sessions when making the real AWS call
-        realRequestHeaders.remove("X-Amz-Security-Token");
-        // replace source host with the real AWS host
-        realRequestHeaders.putSingle("Host", realUri.getHost());
+        MultivaluedMap<String, String> realRequestHeaders = new MultivaluedHashMap<>();
+        request.getRequestHeaders().forEach((key, value) -> {
+            switch (key.toLowerCase()) {
+                case "x-amz-security-token" -> {}   // we don't use sessions when making the real AWS call
+                case "x-amz-date" -> realRequestHeaders.putSingle("X-Amz-Date", signingController.formatRequestInstant(Instant.now())); // use now for the real request
```

Comment:
```suggestion
                case "x-amz-date" -> realRequestHeaders.putSingle("X-Amz-Date", Signer.formatRequestInstant(Instant.now())); // use now for the real request
```

why not make `formatRequestInstant` public in `Signer` so it can be reused?

Reply:
I'd rather not expose `Signer`. Maybe the right thing to do is to move the body of `formatRequestInstant()` into `SigningController`. `Signer` would just expose it's constants statically. wdyt?

Reply:
I made the change - {TAL

Reply:
Yep I like this approach